ketchup/main.py

```diff
@@ -1,121 +1,231 @@
+                "(Example: 15 Aug 2050): "
             )
-    else:
-        print("I'm sorry, but that limit and end date make it impossible!")
-        print("Don't worry, consider these options instead:")
-        print(
-            f"Option 1: You can watch {watch_schedule['base_eps_per_day']} episodes per day to catch up"
-        )
-        if watch_schedule["extra_episodes"]:
-            print(
-                f"You will have {watch_schedule['extra_episodes']} extra,"
-                "you can add an additional watch day or tack them on to one of your other days."
+
+            answer_today = input(
+                "Will you start watching today? [Y/N]: ") or "y"
+
+            if answer_today.lower() == "y":
+                start_date = dt.datetime.today()
+                episode_duration_mins = 24
+                # Set the default episode duration in minutes
+            else:
+                start_date_prompt = "What day will you start?" \
+                                    " (Ex: 15 Aug 2023): "
+                start_date = get_date(start_date_prompt)
+
+            end_date = get_date(end_date_prompt, require_future=True)
+
+            if end_date < start_date:
+                print("The end date cannot be before the start date. "
+                      "Please provide valid dates.\n")
+                continue
+
+            time_available = end_date - start_date
+            min_hours_needed = math.ceil(episodes * episode_duration_mins / 60)
+
+            if time_available.total_seconds() < min_hours_needed * 3600:
+                total_duration_hours = episodes * episode_duration_mins / 60
+                current_time = dt.datetime.now()
+                time_difference = dt.timedelta(hours=total_duration_hours)
+                realistic_end_date = current_time + time_difference
+
+                print(
+                    f"Sorry, it is not possible to watch {episodes} "
+                    f"episodes by that time. "
+                    f"You would instead finish the "
+                    f"show by {realistic_end_date}.\n"
+                )
+                continue
+
+            watch_schedule = wsb.calculate_watch_schedule(
+                episodes,
+                end_date,
+                start_date,
+                episode_duration_mins=episode_duration_mins
             )
-        print(
-            f"Option 2: You could also choose to watch {per_day_limit} episodes per day"
-            "and finish by {watch_schedule['realistic_end_date']}"
-        )
-
-    ff = fun_fact.random_factoid()
-    ff_watch_schedule = wsb.calculate_watch_schedule(
-        ff[1], end_date, start_date, per_day_limit
-    )
-
-    print(
-        f"In comparison you would have to watch {ff_watch_schedule['base_eps_per_day']} episodes per day "
-        "if you were watching {ff[0]}!"
-    )
-    print(" ")
-    print(" ")
-    print(LUFFY)
+
+            if watch_schedule["days"] >= 0:
```

Comment:
no else statement so this would die otherwise?

Reply:
For me it worked without, but I added an else statement that just is set to "continue" now. I receive the same outcome either way, due to the require_future argument.